datastore/sync_entity.go

```diff
@@ -263,6 +293,164 @@ func (dynamo *Dynamo) InsertSyncEntitiesWithServerTags(entities []*SyncEntity) e
 	return nil
 }
 
+// DisableSyncChain marks a chain as disabled so no further updates or commits can happen
```

Comment:
Do we keep track here of the reason why it was disabled (i.e. user requested deletion)?

The spec [lists three possible reasons](https://docs.google.com/document/d/1X7glDHmeIjgipcPWn__S9rSxrI39P9rNw0RmuEuC3BI/edit#heading=h.7vqo8iomb0b3) for a disabled account: expired, deleted, passphrase reset

Only one of these is being implemented right now, but we should probably start keeping track of this data now.